core-common/src/main/java/org/glassfish/jersey/internal/jsr166/SubmissionPublisher.java

```diff
@@ -581,29 +581,45 @@ final int doOffer(long nanos, T item,
     }
 
     /**
-     * Unless already closed, issues {@link
-     * Flow.Subscriber#onComplete() onComplete} signals to current
-     * subscribers, and disallows subsequent attempts to publish.
-     * Upon return, this method does <em>NOT</em> guarantee that all
-     * subscribers have yet completed.
+     * Unless already closed, issues {@link Flow.Subscriber#onComplete()
+     * onComplete} signals to current subscribers when {@code cascading} is
+     * {@code true}. If {@code cascading} is {@code false}, then only closes this
+     * publisher and disallows subsequent attempts to publish. Upon return, even if
+     * {@link Flow.Subscriber#onComplete() onComplete} is called, this method does
+     * <em>NOT</em> guarantee that all subscribers have yet completed.
+     *
+     * @param cascading Controls closing of registered subscribers.
      */
-    public void close() {
+    public void close(boolean cascading) {
```

Comment:
This is jdk 9 SubmissionPublisher. It does not have close(boolean). Should we add it?